### PR TITLE
Better error messages

### PR DESCRIPTION
**Pull request by drewbanin**
Better error messages for https://github.com/fishtown-analytics/dbt/issues/294

#### No `profile` specified in `dbt_project.yml` or provided on CLI. No `profiles.yml` file
```
$ dbt run
Encountered an error while reading the project:
dbt cannot run because no profile was specified for this dbt project.
To specify a profile for this project, add a line like the this to
your dbt_project.yml file:

profile: [profile name]

Here, [profile name] should be replaced with a profile name
defined in your profiles.yml file. You can find profiles.yml here:

/Users/drew/.dbt/profiles.yml

There are no profiles defined in your profiles.yml file

For more information on configuring profiles, please consult the dbt docs:

https://dbt.readme.io/docs/configure-your-profile

Encountered an error:
Could not run dbt
```

#### `profiles.yml` not found (but a profile is specified in `dbt_project.yml`)
```
$ dbt run
Encountered an error while reading the project:
Could not find profile named 'default'
There are no profiles defined in your profiles.yml file

For more information on configuring profiles, please consult the dbt docs:

https://dbt.readme.io/docs/configure-your-profile

Encountered an error:
Could not run dbt
```

#### `profiles.yml` contains invalid yaml
```
$ dbt run
Encountered an error:

dbt encountered an error while trying to read your profiles.yml file.

Syntax error near line 11
------------------------------
8  |   target: dev
9  |   outputs:
10 |     dev
11 |       type: postgres
12 |       host: localhost
13 |       user: drew

Raw Error:
------------------------------
mapping values are not allowed here
  in "<unicode string>", line 11, column 11:
          type: postgres
              ^
```

#### Invalid yaml in a `schema.yml` file:
```
$ dbt test
Error reading package_name:base/schema.yml - Skipping
Syntax error near line 16
------------------------------
13 |         not_null
14 |             - now
15 |
16 |         unique:
17 |             - now
18 |

Raw Error:
------------------------------
mapping values are not allowed here
  in "<unicode string>", line 16, column 15:
            unique:
                  ^
```

@cmcarthur let me know if you think i went overboard here...

### CODE REVIEW

File: dbt/clients/yaml_helper.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    return "\n".join(output)
+
+
+def load_yaml_text(contents):
+    try:
+        return yaml.safe_load(contents)
+    except (yaml.scanner.ScannerError, yaml.YAMLError) as e:
+        if hasattr(e, 'problem_mark'):
+            error = contextualized_yaml_erro(contents, e)
```
Comment (cmcarthur) [inline dbt/clients/yaml_helper.py]: contextualized_yaml_erro**r**?

File: dbt/clients/yaml_helper.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        else:
+            error = dbt.compat.to_string(e)
+
+        raise dbt.exceptions.ValidationException(error)
```
Comment (cmcarthur) [inline dbt/clients/yaml_helper.py]: this is great, but the code here is a little rough around the edges.

File: dbt/clients/yaml_helper.py (hunk 1 above)
Comment (drewbanin) [inline dbt/clients/yaml_helper.py]: 🤦‍♂️

**Comment by drewbanin** [comment]
@cmcarthur good points -- i think this is ready for another look

**Pull request merged**